clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedInternalCommandConsumer.java

```
@@ -393,46 +392,47 @@ void handleCommandMessage(final KafkaConsumerRecord<String, Buffer> record) {
-                    if (lastHandledOffset != null && commandOffset <= lastHandledOffset) {
-                        LOG.debug("ignoring command - record partition offset {} <= last handled offset {} [{}]", commandOffset,
-                                lastHandledOffset, command);
-                        TracingHelper.logError(currentSpan, "command record already handled before");
-                        commandContext.release(new CommandAlreadyProcessedException());
-                    } else {
-                        lastHandledPartitionOffsets.put(commandPartition, commandOffset);
-                        LOG.trace("using [{}] for received command [{}]", commandHandler, command);
-                        // command.isValid() check not done here - it is to be done in the command handler
-                        commandHandler.handleCommand(commandContext);
-                    }
-                } else {
-                    LOG.info("no command handler found for command [{}]", command);
-                    commandContext.release(new NoConsumerException("no command handler found for command"));
-                }
-            });
+        // partition index and offset here are related to the *tenant-based* topic the command was originally received in
+        // therefore they are stored with the tenant as key
+        final Map<Integer, Long> lastHandledPartitionOffsets = lastHandledPartitionOffsetsPerTenant
+                .computeIfAbsent(command.getTenant(), k -> new HashMap<>());
+        final Long lastHandledOffset = lastHandledPartitionOffsets.get(commandPartition);
+        if (lastHandledOffset != null && commandOffset <= lastHandledOffset) {
+            LOG.debug("ignoring command - record partition offset {} <= last handled offset {} [{}]", commandOffset,
+                    lastHandledOffset, command);
+            TracingHelper.logError(currentSpan, "command record already handled before");
```

is this an error? I would assume we can simply silently ignore the duplicate or can't we?

---

I've removed this line and the following here, also not creating a span for this case.